### PR TITLE
Pin mimesis version in setup.py.

### PR DESCRIPTION
**Pull request by bdice**
## Description
The dependency pinning for `mimesis` in cudf's `setup.py` didn't match the conda environment. It was missing a pinning to `<4.1` from #8745. However, based on the conversation in #8551, this pinning of `<4.1` was only chosen because 4.1.0 wasn't yet available on conda-forge. Since the current version of mimesis is now 6.1.1, this PR updates the mimesis pinning to `>=4.1` and uses `generate_string` instead of `schoice`. I tested this locally with mimesis 6.1.1 and mimesis 4.1.0 and both passed tests.

Merge this PR concurrently with https://github.com/rapidsai/integration/pull/547.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by vyasr** [comment]
Opened rapidsai/integration#548 to unblock tests.

**Comment by galipremsagar** [comment]
> Opened [rapidsai/integration#548](https://github.com/rapidsai/integration/pull/548) to unblock tests.

Looks like @bdice already opened a PR: https://github.com/rapidsai/integration/pull/547

**Comment by galipremsagar** [comment]
rerun tests

**Comment by bdice** [comment]
rerun tests

**Review by ajschmidt8 — APPROVED** [review]
Approving `ops-codeowner` file changes

**Comment by galipremsagar** [comment]
@gpucibot merge
@gpucibot merge

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**
